src/platform/web/config.ts

```diff
@@ -76,6 +77,18 @@ const Config: IPlatformConfig = {
       crypto.getRandomValues(arr);
     };
   })(globalObject.crypto),
+  getRandomArrayBuffer: (function (crypto) {
+    const DEFAULT_BLOCKLENGTH_WORDS = 4; // 32-bit words
```

Comment:
I'm a bit confused by this implementation — what's the idea of this "32-bit word" thing, and why do we have to go via `BufferUtils`? I haven't looked into the exact API calls needed but I assumed that something along the lines of the following would suffice:

1. create a `Int8Array` of the desired length
1. pass that to `crypto.getRandomValues`
1. return the underlying `ArrayBuffer`

Reply:
Oh, I see now that it's been copied and pasted from the web crypto code. If you want to keep the implementation as it is, then I’d suggest one of:

1. creating an issue to revisit this implementation and putting a comment about it here, because it’s going to confuse people in the future I think, or 
2. at least mentioning in the commit message that this is copied from existing code

Reply:
> Oh, I see now that it's been copied and pasted from the web crypto code.

Yeah, this is just a copy paste.
I actually wanted to further change this code after unifying it all under `IPlatformConfig.getRandomArrayBuffer`, but I think I kinda lost it with all other changes.

If you look closely at those lines:

```javascript
const DEFAULT_BLOCKLENGTH_WORDS = 4; // 32-bit words

return async function (byteLength: number): Promise<ArrayBuffer> {
  const blockRandomArray = new Uint32Array(DEFAULT_BLOCKLENGTH_WORDS);
  const words = byteLength / 4;
  const nativeArray = words === DEFAULT_BLOCKLENGTH_WORDS ? blockRandomArray : new Uint32Array(words);

  ...
```

You will notice that `const nativeArray = words === DEFAULT_BLOCKLENGTH_WORDS ? blockRandomArray : new Uint32Array(words);` doesn't actually accomplish anything. It's the same if we were always creating Uint32Array with `words` length in the first place.

It also didn't do anything in web crypto code. Before https://github.com/ably/ably-js/pull/1583/commits/fccd4ac950858634ccb7e0120ad5520261be63b8 it would actually have different behavior, since we were reusing pre-existing array, but after that change there is no point in `words === DEFAULT_BLOCKLENGTH_WORDS` check anymore.

So it can reduced to:

```javascript
return async function (byteLength: number): Promise<ArrayBuffer> {
  const words = byteLength / 4;
  const nativeArray = new Uint32Array(words);

  crypto.getRandomValues(nativeArray);
  return bufferutils.toArrayBuffer(nativeArray);
};
```

And now there is really now point to use `words` and Uint32Array anymore, so per your first comment we can change it to:

```javascript
return async function (byteLength: number): Promise<ArrayBuffer> {
  const nativeArray = new Uint8Array(byteLength);
  crypto.getRandomValues(nativeArray);
  return nativeArray.buffer;
};
```

Will add those changes with explanation in commit message.

Reply:
Simplified this function and added explanation to commit message in https://github.com/ably/ably-js/pull/1629/commits/5106b006944d06e66adc477f66e158a4752d70f1.